lib/upload_creator.rb

```diff
@@ -123,7 +123,7 @@ def create_for(user_id)
 
       # store the file and update its url
       File.open(@file.path) do |f|
-        url = Discourse.store.store_upload(f, @upload)
+        url, @upload.etag = Discourse.store.store_upload(f, @upload)
```

Comment:
I'm curious what happens in this assignment when Discourse.store is a different store? The pattern that we're using now requires all new devs to be aware of the fact that only the s3 store will return an etag.

Reply:
It's not a problem. If etag is not returned then it will be assigned to `nil` automatically. Anyway I improved the code further.

Reply:
Can we set the etag in the `S3Store` instead? Since we already pass the `@upload` to `Discourse.store.store_upload`, perhaps we can set the etag within that method and only return the `url` only as it used to?

Reply:
@tgxworld now it's done like that.